src/main/scala/com/typesafe/sbt/packager/archetypes/JavaServerApplication.scala

```diff
@@ -59,27 +57,26 @@ object JavaServerAppPackaging extends AutoPlugin {
     },
     // === etc config mapping ===
     bashScriptEnvConfigLocation := Some("/etc/default/" + (packageName in Linux).value),
-    linuxEtcDefaultTemplate <<= sourceDirectory map { dir =>
-      val overrideScript = dir / "templates" / ETC_DEFAULT
-      if (overrideScript.exists) overrideScript.toURI.toURL
-      else etcDefaultTemplateSource
-    },
-    linuxStartScriptName := None,
-    makeEtcDefault <<= (packageName in Linux, target in Universal, linuxEtcDefaultTemplate, linuxScriptReplacements)
-      map makeEtcDefaultScript,
-    linuxPackageMappings <++= (makeEtcDefault, bashScriptEnvConfigLocation) map { (conf, envLocation) =>
-      val mapping = for (
-        path <- envLocation;
-        c <- conf
-      ) yield LinuxPackageMapping(Seq(c -> path), LinuxFileMetaData(Users.Root, Users.Root, "644")).withConfig()
+    // (serverLoading in Rpm) apply getStartScriptLocation
 
-      mapping.toSeq
-    }
+    linuxStartScriptName := None
+  )
 
+  /* etcDefaultConfig is dependent on serverLoading (systemd, systemv, etc.),
+   * and is therefore distro specific. As such, these settings cannot be defined
+   * in the global config scope. */
+  private val etcDefaultConfig: Seq[Setting[_]] = Seq(
+    linuxEtcDefaultTemplate <<= (sourceDirectory, serverLoading)
+      map getEtcTemplateSource,
```

Comment:
We try to avoid the sbt symbolic operators. Instead of
```
linuxEtcDefaultTemplate <<= (sourceDirectory, serverLoading) map getEtcTemplatSource
```

we try to clean up the code base using

```
linuxEtcDefaultTemplate := getEtcTemplatSource(sourceDirectory.value, serverLoading.value)
```

In some cases this adds a few extra stuff, but makes it easier to understand for newcomers.